topics/topic3/shopping-cart/src/test/java/shoppingcart/ShoppingCartTest.java

```diff
@@ -0,0 +1,53 @@
+package shoppingcart;
+
+import java.util.ArrayList;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ShoppingCartTest {
+
+	ShoppingCart testShoppingCart = new ShoppingCart();
+	Product newProduct = new Product(0, "name");
+
+	@Test
+	public void whenOneProductIsAddedThenReturnTheTotalProductsOfCartPlusOne() {
+		int totalProducts = testShoppingCart.totalProducts();
+		testShoppingCart.addProdcut(newProduct);
+		Assert.assertEquals(totalProducts + 1, testShoppingCart.totalProducts());
```

Comment:
you checked the amount of the items, you missed check newProduct.
Assert.assertEquals(totalProducts.get(0), newProduct)